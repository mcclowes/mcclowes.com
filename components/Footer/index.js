import React from "react";

import styled from "styled-components/macro";

const Container = styled.footer`
  width: 100%;
  height: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 1em;
`;

const Footer = () => {
  return (
    <>
      <Container>
        <a href="mailto:contact@mcclowes.com">contact@mcclowes.com</a>
      </Container>
    </>
  );
};

export default Footer;
