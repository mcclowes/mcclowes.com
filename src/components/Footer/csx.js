import styled from "styled-components";

export const FooterLinks = styled.div`
  background: ${props=>props.theme.colors.black};
  padding: 1rem;
  padding-left: 2.5em;

  a {
    color: white;
    text-decoration: none;
    font-family: helvetica;
    font-weight: bolder;
    letter-spacing: 0.3rem;
    text-transform: uppercase;
    text-decoration: none;
    font-size: 1.2rem;

    &,
    &:hover,
    &:active {
      -webkit-text-stroke-color: ${props=>props.theme.colors.primary};
      -webkit-text-stroke-width: 1px;
      color: transparent;
    }

    &:hover { opacity: 0.75; }
    &:active { opacity: 0.5; }
  }
`;