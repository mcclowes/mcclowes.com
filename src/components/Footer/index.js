import React from "react";
import PropTypes from "prop-types";
import Helmet from "react-helmet";
import moment from "moment";

import Link from "gatsby-link";

import { Container, Grid, Header, List, Segment, } from "semantic-ui-react";

// ----------------------------------------------------

// ----------------------------------------------------

const Footer = props => (
	<Segment color = "grey" style = { { padding: "5em 0em", } } inverted vertical>
		<Container>
			<Grid divided inverted stackable>
				<Grid.Row>
					@mcclowes
				</Grid.Row>
			</Grid>
		</Container>
	</Segment>
);

export default Footer;
