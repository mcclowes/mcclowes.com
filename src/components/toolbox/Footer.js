import React from "react";
import styled from "styled-components";

import * as mixins from "../style/mixins";
import * as vars from "../style/vars";

import { Icon, } from "./";

import data from "src/data.js";

// --------------------------------------------------

const Wrapper = styled.footer`
	background-color: ${R.path([ "theme", "footer", ])};
	${ mixins.bpEither("height", vars.dim.footer.height) }
	${ mixins.bpEither("padding", vars.dim.nav.margin) };
	align-items: center;
	//border-top: 1px solid ${ mixins.tr(0.2) };
	bottom: 0;
	display: flex;
	justify-content: space-between;
	left: 0;
	overflow: hidden;
	position: absolute;
	right: 0;
`;

const Left = styled.div`
	font-weight: bold;
	color: ${R.path([ "theme", "text", ])};

	a {
		color: ${R.path([ "theme", "text", ])};

		&:hover,
		&:active {
			color: #eee;
		}
	}
`;

const Right = styled.div`
	display: flex;
	flex-direction: row;
	font-size: 1.5em;

	a {
		color: ${R.path([ "theme", "text", ])};
		margin-left: 0.5em;

		&:hover,
		&:active {
			color: #eee;
		}
	}
`;

const Footer = () =>
	<Wrapper>
		<Left>Site by <a href="https://codogo.io">Codogo</a></Left>

		<Right>
			<a href = { `https://www.twitter.com/${ data.twitterUsername }` }><Icon type = "twitter"/></a>

			<a href = { `https://www.medium.com/@${ data.mediumUsername }` }><Icon type = "medium"/></a>

			<a href = { `https://www.linkedin.com/in/${ data.linkedInUsername }` }><Icon type = "linkedin"/></a>

			<a href = { `https://www.github.com/${ data.githubUsername }` }><Icon type = "github"/></a>
		</Right>
	</Wrapper>;

export default Footer;
