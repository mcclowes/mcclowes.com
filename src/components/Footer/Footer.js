import Link from "gatsby-link";
import PropTypes from "prop-types";
import React from "react";
import { FooterLinks } from './csx'

const Footer = ({ siteTitle, }) => (
	<FooterLinks>
    <a href="https://personal-sites-web-ring.freddieridell.now.sh/prev">⬅️</a>
    <a href="https://personal-sites-web-ring.freddieridell.now.sh/">Webring</a>
    <a href="https://personal-sites-web-ring.freddieridell.now.sh/next">➡️</a>
    <a href="https://personal-sites-web-ring.freddieridell.now.sh/rand">🔀</a>
  </FooterLinks>
);

Footer.propTypes = {
	siteTitle: PropTypes.any,
};

export default Footer;
