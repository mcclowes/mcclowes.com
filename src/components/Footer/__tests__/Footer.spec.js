import Footer from "../Footer";
import React from "react";
import { shallow } from "enzyme";

describe("Footer", () => {
  it("renders component", () => {
    const wrapper = shallow(<Footer siteTitle="Default Title" />);

    expect(wrapper).toMatchSnapshot();
  });
});
