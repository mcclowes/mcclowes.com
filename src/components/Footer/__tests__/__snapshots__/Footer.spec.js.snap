// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders component 1`] = `
<csx__FooterLinks>
  <a
    href="https://personal-sites-web-ring.freddieridell.now.sh/prev"
  >
    ⬅️
  </a>
  <a
    href="https://personal-sites-web-ring.freddieridell.now.sh/"
  >
    Webring
  </a>
  <a
    href="https://personal-sites-web-ring.freddieridell.now.sh/next"
  >
    ➡️
  </a>
  <a
    href="https://personal-sites-web-ring.freddieridell.now.sh/rand"
  >
    🔀
  </a>
</csx__FooterLinks>
`;
